pkg/autodiscovery/autoconfig.go

```diff
@@ -500,39 +570,53 @@ func (ac *AutoConfig) processRemovedConfigs(removedConfigs []integration.Config)
 		if config.IsTemplate() {
 			ac.templateCache.Del(config)
 		}
+		log.Infof("Removed %q", config.Name)
 	}
 }
 
 // collect is just a convenient wrapper to fetch configurations from a provider and
 // see what changed from the last time we called Collect().
-func (ac *AutoConfig) collect(pd *providerDescriptor) (new, removed []integration.Config) {
-	new = []integration.Config{}
-	removed = []integration.Config{}
+func (ac *AutoConfig) collect(pd *providerDescriptor) ([]integration.Config, []integration.Config) {
+	var newConf []integration.Config
+	var removedConf []integration.Config
+	old := pd.configs
 
 	fetched, err := pd.provider.Collect()
 	if err != nil {
 		log.Errorf("Unable to collect configurations from provider %s: %s", pd.provider, err)
-		return
+		return nil, nil
 	}
 
 	for _, c := range fetched {
+		log.Infof("Fetched %s", c)
 		if !pd.contains(&c) {
-			new = append(new, c)
+			log.Infof("New config: %s", c)
+			newConf = append(newConf, c)
+			continue
+		}
+		// Check the freshness of c. Reschedule if necessary.
+		if tagger.OutdatedTags(c.ADIdentifiers) {
```

Comment:
ADIdentifiers and entity names happen to follow the same format, but not all AD identifiers are valid tagger entities (docker short/long image name). Couldn't we get a valid entity name from the listeners instead?